woocommerce/woocommerce-admin
# Replace select controls with new SelectControl (Autocomplete) component

Fixes (partially) #2834

* Adds the `isSearchable` prop to the autocomplete component and renames it to `SelectControl` to cover more use cases.
* Allows strings for simple selects `selected` values in addition to arrays.
* Replaces instances of `SimpleSelectControl` with this component.
* Replaces the country dropdown with this component.

### Screenshots
<img width="487" alt="Screen Shot 2019-10-03 at 1 58 29 AM" src="https://user-images.githubusercontent.com/10561050/66113647-0db53780-e582-11e9-9fc4-6b837ba38e17.png">
<img width="541" alt="Screen Shot 2019-10-03 at 1 51 14 AM" src="https://user-images.githubusercontent.com/10561050/66113649-0db53780-e582-11e9-9c36-ab4a5abdb5aa.png">

### Detailed test instructions:

1. Enable the onboarding profiler and visit both store details and business details.
2. Make sure the country dropdown and all dropdowns on the business details step continue to work as expected.
3.  Make sure that submitted/changed values for dropdowns continue working as expected.

### Changelog Note:

Dev: Replace applicable select components with searchable SelectControl component.
